src/services/redisStorageUtils.ts

```diff
@@ -51,4 +51,17 @@ export default class RedisStorageUtils {
       return left(error);
     }
   }
+
+  protected genericPromisifyCallback<E, T>(
```

Comment:
probably this should go into a generic utils module (not in redisStorageUtils.ts).
consider also using `taskify` . https://github.com/gcanti/fp-ts/blob/f8725946ab9e2b7bd05dbf015007955515b69c3a/docs/modules/TaskEither.ts.md#taskify-function

which returns a `TaskEither`  (you have to call `run()` on it)

Reply:
I use a more generic implementation in a specific util file to promisify generic async function, compatible with redis. What you think about it?
Added in commit 824b0e1

Reply:
I'll attempt to use `taskify` first which has a semantic simlar to Either and does not require extra code, ie:

```ts
const mget = taskify(this.redisClient.mget.bind(this.redisClient));
const errorOrResult = await mget(<params>) .run();
```

or directly:

```ts
return await mget(<params>) .map(result => ...).mapLeft(err => ...).run();
```

If you get issue with the types see: https://github.com/gcanti/fp-ts/pull/621
(or ask @denisfrezzato : - )

Reply:
@gunzip `taskify` is available like `parseJSON` from `v2.0.0` of `fp-ts`

Reply:
I try like `parseJSON` to backport this function

Reply:
it looks that taskify is in fp-ts 1.12.0:

https://github.com/gcanti/fp-ts/blob/1.12.0/src/TaskEither.ts#L362

Reply:
@gunzip look if it's this the right way. Fixed in 2e3769f